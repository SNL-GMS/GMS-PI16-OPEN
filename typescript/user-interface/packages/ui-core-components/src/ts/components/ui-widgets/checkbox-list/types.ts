import { Position } from '@blueprintjs/core';
import { IconName } from '@blueprintjs/icons';
import Immutable from 'immutable';

export interface CheckboxListProps {
  checkboxEnum: any;
  enumToCheckedMap: Map<any, boolean>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumToColorMap?: Map<any, string>;
  onChange(map: Map<any, boolean>);
}

export interface SimpleCheckboxListProps {
  checkBoxListEntries: CheckboxListEntry[];
  onChange?(name: string): void;
}

export interface SimpleCheckboxListState {
  checkboxEntriesMap: Immutable.Map<string, CheckboxListEntry>;
}

export interface CheckboxListEntry {
  name: string;
  color?: string;
  iconButton?: IconButton;
  isChecked: boolean;
}

export interface IconButton {
  iconName: IconName;
  popover?: {
    position: Position;
    usePortal?: boolean;
    minimal?: boolean;
    content?: string | JSX.Element;
  };
  onClick?(event: React.MouseEvent<HTMLButtonElement>): void;
}
