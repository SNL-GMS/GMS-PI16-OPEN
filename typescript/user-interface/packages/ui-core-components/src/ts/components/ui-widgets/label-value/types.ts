import React from 'react';

/**
 * app determines the default styling for the label-value toolbar item, defaults to soh
 */
export interface LabelValueProps {
  value: string;
  label: string;
  tooltip: string;
  valueColor?: string;
  ianApp?: boolean;
  styleForValue?: React.CSSProperties;
  containerClass?: string;
}
