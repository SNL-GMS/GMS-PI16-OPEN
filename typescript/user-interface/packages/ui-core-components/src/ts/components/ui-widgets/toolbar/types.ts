import { IconName } from '@blueprintjs/core';
import { DateTimeFormat } from '@gms/common-util';
import React from 'react';

import { DateRangePickerTypes } from '../date-range-picker';

export interface ToolbarProps {
  items: ToolbarItem[];
  toolbarWidthPx: number;
  itemsLeft?: ToolbarItem[];
  minWhiteSpacePx?: number;
  spaceBetweenItemsPx?: number;
  hidden?: boolean;
  overflowIcon?: IconName;
}
export interface ToolbarState {
  indicesToOverflow: number[];
  leftIndicesToOverflow: number[];
  whiteSpaceAllotmentPx: number;
  checkSizeOnNextDidMountOrDidUpdate: boolean;
}

export type ToolbarItem =
  | NumericInputItem
  | DropdownItem
  | IntervalPickerItem
  | DateRangePickerItem
  | PopoverItem
  | SwitchItem
  | ButtonItem
  | ButtonGroupItem
  | LabelValueItem
  | CheckboxDropdownItem
  | LoadingSpinnerItem
  | CustomItem;

interface ToolbarItemBase {
  label?: string;
  tooltip: string;
  tooltipForIssue?: string;
  type: ToolbarItemType;
  rank: number;
  style?: React.CSSProperties;
  widthPx?: number;
  labelRight?: string;
  disabled?: boolean;
  icon?: IconName;
  onlyShowIcon?: boolean;
  menuLabel?: string;
  cyData?: string;
  hasIssue?: boolean;
  ianApp?: boolean; // If true, use default ian styling. defaults to false (SoH)
  onMouseEnter?(): void;
  onMouseOut?(): void;
}

interface NumericInputItem extends ToolbarItemBase {
  value: number;
  minMax: MinMax;
  step?: number;
  requireEnterForOnChange?: boolean;
  onChange(value: number);
}

interface DropdownItem extends ToolbarItemBase {
  dropdownOptions: any;
  dropdownText?: any;
  value: any;
  custom?: boolean;
  displayLabel?: boolean;
  onChange(value: any);
}

interface IntervalPickerItem extends ToolbarItemBase {
  startDate: Date;
  endDate: Date;
  defaultIntervalInHours: number;
  shortFormat?: boolean;
  onChange(startDate: Date, endDate: Date);
  onApplyButton(startDate: Date, endDate: Date);
}

interface DateRangePickerItem extends ToolbarItemBase {
  startTimeMs: number;
  endTimeMs: number;
  format: DateTimeFormat;
  durations?: DateRangePickerTypes.DurationOption[];
  minStartTimeMs?: number;
  maxEndTimeMs?: number;
  onChange(startTimeMs: number, endTimeMs: number);
  onApplyButton(startTimeMs: number, endTimeMs: number);
}

interface PopoverItem extends ToolbarItemBase {
  popoverContent: JSX.Element;
  onPopoverDismissed();
}

interface SwitchItem extends ToolbarItemBase {
  value: boolean;
  onChange(value: boolean);
}

interface ButtonItem extends ToolbarItemBase {
  onClick();
}

interface ButtonGroupItem extends ToolbarItemBase {
  buttons: ButtonItem[];
}

interface LabelValueItem extends ToolbarItemBase {
  value: string;
  valueColor?: string;
  styleForValue?: React.CSSProperties;
}

interface CheckboxDropdownItem extends ToolbarItemBase {
  values: Map<any, boolean>;
  colors?: Map<any, string>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumOfKeys: any;
  onChange(value: any): void;
  onPopUp?(ref?: HTMLDivElement): void;
  onPopoverDismissed?(): void;
}

interface LoadingSpinnerItem extends ToolbarItemBase {
  itemsToLoad: number;
  itemsLoaded?: number;
  hideTheWordLoading?: boolean;
  hideOutstandingCount?: boolean;
}

interface CustomItem extends ToolbarItemBase {
  element: JSX.Element;
}

export interface MinMax {
  min: number;
  max: number;
}

export enum ToolbarItemType {
  Switch = 'Switch',
  Popover = 'Popover',
  Dropdown = 'Dropdown',
  NumericInput = 'NumericInput',
  Button = 'Button',
  IntervalPicker = 'IntervalPicker',
  DateRangePicker = 'DateRangePicker',
  ButtonGroup = 'ButtonGroup',
  LabelValue = 'LabelValue',
  CheckboxList = 'CheckboxList',
  LoadingSpinner = 'LoadingSpinner',
  CustomItem = 'CustomItem'
}

export {
  ButtonGroupItem,
  ButtonItem,
  CheckboxDropdownItem,
  CustomItem,
  DateRangePickerItem,
  DateRangePickerTypes,
  DropdownItem,
  IntervalPickerItem,
  LabelValueItem,
  LoadingSpinnerItem,
  NumericInputItem,
  PopoverItem,
  SwitchItem
};
