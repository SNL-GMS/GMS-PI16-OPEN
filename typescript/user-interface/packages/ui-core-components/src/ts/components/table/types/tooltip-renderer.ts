/* eslint-disable @typescript-eslint/no-empty-interface */
import { ITooltipParams as AgITooltipParams } from 'ag-grid-community';

/** Wrapper interface class around ag-grid interface `ITooltipParams` */
export interface TooltipParams extends AgITooltipParams {}
