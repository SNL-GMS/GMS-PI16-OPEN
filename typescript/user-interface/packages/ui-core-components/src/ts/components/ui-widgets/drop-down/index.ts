// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { DropDown } from './drop-down';

export { DropDown };
