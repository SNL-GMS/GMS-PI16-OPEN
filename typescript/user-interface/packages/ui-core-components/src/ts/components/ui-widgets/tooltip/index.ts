import { TooltipWrapper, WithTooltip } from './tooltip';

export { TooltipWrapper, WithTooltip };
