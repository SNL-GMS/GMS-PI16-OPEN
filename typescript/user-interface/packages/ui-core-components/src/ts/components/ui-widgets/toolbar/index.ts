// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { ToolbarComponent } from './toolbar-component';
import * as ToolbarTypes from './types';

export { ToolbarComponent as Toolbar, ToolbarTypes };
