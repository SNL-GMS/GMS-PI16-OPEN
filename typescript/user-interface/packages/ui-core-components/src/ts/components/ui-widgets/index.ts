import { CheckboxListEntry } from './checkbox-list';
import { CheckboxSearchList, CheckboxSearchListTypes } from './checkbox-search-list';
import { DateRangePicker, DateRangePickerTypes } from './date-range-picker';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { DropDown } from './drop-down';
import { FilterableOptionList, FilterableOptionListTypes } from './filterable-option-list';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { Form, FormTypes } from './form';
import { HistoryList, HistoryListTypes } from './history-list';
import { IntervalPicker, IntervalPickerTypes } from './interval-picker';
import { LabelValue } from './label-value';
import { LoadingSpinner } from './loading-spinner';
import { PopoverButton } from './popover-button';
import { TextArea } from './text-area';
import { TimePicker, TimePickerTypes } from './time-picker';
import { TitleBar, TitleBarTypes } from './title-bar';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { Toolbar, ToolbarTypes } from './toolbar';
import { TooltipWrapper, WithTooltip } from './tooltip';
import { Widget, WidgetTypes } from './widgets';

export { CheckboxSearchList, CheckboxSearchListTypes };
export { CheckboxListEntry };
export { SimpleCheckboxList } from './checkbox-list/simple-checkbox-list';
export { DropDown };
export { FilterableOptionList, FilterableOptionListTypes };
export { Form, FormTypes };
export { PopoverButton };
export { HistoryList, HistoryListTypes };
export { TimePicker, TimePickerTypes };
export { DateRangePicker, DateRangePickerTypes };
export { IntervalPicker, IntervalPickerTypes };
export { LabelValue };
export { LoadingSpinner };
export { TextArea };
export { Toolbar, ToolbarTypes };
export { TitleBar, TitleBarTypes };
export { Widget, WidgetTypes };
export { TooltipWrapper, WithTooltip };
