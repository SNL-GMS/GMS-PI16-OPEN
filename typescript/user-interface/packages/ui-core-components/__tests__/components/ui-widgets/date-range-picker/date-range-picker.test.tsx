/* eslint-disable @typescript-eslint/no-magic-numbers */
import {
  DATE_FORMAT,
  DATE_TIME_FORMAT,
  DATE_TIME_FORMAT_WITH_FRACTIONAL_SECOND_PRECISION,
  DATE_TIME_FORMAT_WITH_SECOND_PRECISION
} from '@gms/common-util';
import React from 'react';

import { DateRangePicker } from '../../../../src/ts/components/ui-widgets/date-range-picker';
import { DateRangePickerProps } from '../../../../src/ts/components/ui-widgets/date-range-picker/types';

jest.useFakeTimers();

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

const MOCK_TIME = 1609506000000;
const MOCK_TIME_STR = '2021-01-01 13:00:00';

const testDate: Date = new Date(MOCK_TIME);

const mockDate: any = new Date(MOCK_TIME);
mockDate.now = () => MOCK_TIME;
mockDate.getMonth = () => testDate.getMonth();
mockDate.getFullYear = () => testDate.getFullYear();
mockDate.getDate = () => testDate.getDate();
mockDate.getMinutes = () => testDate.getMinutes();
Date.constructor = jest.fn(() => mockDate);
jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
Date.now = jest.fn(() => MOCK_TIME);
Date.UTC = jest.fn(() => MOCK_TIME);

jest.mock('moment', () => {
  // mock chain builder pattern
  const mMoment = {
    ...jest.requireActual('moment'),
    utc: jest.fn(() => mMoment),
    format: jest.fn(() => MOCK_TIME_STR),
    default: {
      utc: jest.fn(() => mMoment)
    }
  };

  // mock the constructor and to modify instance methods
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const fn: any = jest.fn(() => {
    mMoment.format = jest.fn(() => MOCK_TIME_STR);
    mMoment.default = {
      utc: jest.fn(() => mMoment)
    };
    return mMoment;
  });

  // mock moment methods that depend on moment not on a moment instance
  fn.unix = () => ({ utc: () => mMoment });
  fn.utc = jest.fn(() => mMoment);
  return fn;
});

// eslint-disable-next-line import/no-deprecated
const lodash = jest.requireActual('lodash');
lodash.uniqueId = () => '1';

describe('DateRangePicker', () => {
  it('should be defined', () => {
    expect(DateRangePicker).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(mockDate).toMatchSnapshot();

    const component = Enzyme.shallow(
      <DateRangePicker
        durations={[
          {
            description: 'one second',
            value: 1000
          },
          {
            description: 'two seconds',
            value: 2000
          }
        ]}
        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
        startTimeMs={MOCK_TIME - 2000}
        endTimeMs={MOCK_TIME}
        onNewInterval={jest.fn()}
        onApply={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();

    component.update();
    component.setProps({
      // eslint-disable-next-line @typescript-eslint/no-magic-numbers
      startTimeMs: MOCK_TIME - 3000,
      endTimeMs: MOCK_TIME
    });
    component.update();
    expect(component).toMatchSnapshot();
    component.instance().onApply();
    expect(component).toMatchSnapshot();

    component.setProps({
      format: DATE_TIME_FORMAT_WITH_FRACTIONAL_SECOND_PRECISION
    });
    component.update();
    expect(component).toMatchSnapshot();

    component.setProps({
      format: DATE_TIME_FORMAT_WITH_SECOND_PRECISION
    });
    component.update();
    expect(component).toMatchSnapshot();

    component.setProps({
      format: DATE_FORMAT,
      minStartTimeMs: MOCK_TIME - 8000,
      maxEndTimeMs: MOCK_TIME
    });
    component.update();
    expect(component).toMatchSnapshot();
  });

  it('button opens the popup', () => {
    const component = Enzyme.mount(
      <DateRangePicker
        durations={[
          {
            description: 'one second',
            value: 1000
          },
          {
            description: 'two seconds',
            value: 2000
          }
        ]}
        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
        startTimeMs={MOCK_TIME - 2000}
        endTimeMs={MOCK_TIME}
        onNewInterval={jest.fn()}
        onApply={jest.fn()}
      />
    );

    let popup = component.find('DateRangePopup');
    expect(popup.prop('isOpen')).toBeFalsy();

    const button = component.find({ 'data-cy': 'date-range-picker-edit' }).first();
    button.simulate('click');

    popup = component.find('DateRangePopup');
    expect(popup.prop('isOpen')).toBeTruthy();

    const button2 = component.find({ 'data-cy': 'modal-cancel-button' }).first();
    button2.simulate('click');
    popup = component.find('DateRangePopup');
    expect(popup.prop('isOpen')).toBeFalsy();
  });

  it('can test DateRangePicker functions', () => {
    const onNewInterval = jest.fn();
    const onApply = jest.fn();
    const props: DateRangePickerProps = {
      durations: [
        {
          description: 'one second',
          value: 1000
        },
        {
          description: 'two seconds',
          value: 2000
        }
      ],
      // eslint-disable-next-line @typescript-eslint/no-magic-numbers
      startTimeMs: MOCK_TIME - 2000,
      endTimeMs: MOCK_TIME,
      format: DATE_TIME_FORMAT,
      onNewInterval,
      onApply
    };

    // eslint-disable-next-line react/jsx-props-no-spreading
    let component = Enzyme.mount(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <DateRangePicker {...props} onApply={undefined} onNewInterval={undefined} />
    );
    let picker: any = component.find(DateRangePicker).instance();
    picker.onApply(MOCK_TIME - 2000, MOCK_TIME);
    jest.runAllTimers();
    expect(onApply).toBeCalledTimes(0);

    // eslint-disable-next-line react/jsx-props-no-spreading
    component = Enzyme.mount(<DateRangePicker {...props} />);
    picker = component.find(DateRangePicker).instance();
    picker.onApply(MOCK_TIME - 2000, MOCK_TIME);
    jest.runAllTimers();
    expect(onApply).toBeCalledTimes(1);
  });
});
