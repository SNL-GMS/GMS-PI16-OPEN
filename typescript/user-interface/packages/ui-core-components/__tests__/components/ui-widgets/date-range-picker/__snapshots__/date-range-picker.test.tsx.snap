// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker matches the snapshot 1`] = `2021-01-01T13:00:00.000Z`;

exports[`DateRangePicker matches the snapshot 2`] = `
<div
  className="date-range-picker"
  id="date-range-picker-id-1"
>
  <Component
    className="bp3-label bp3-inline date-range-picker__label"
  >
    Time Range: 2021-01-01 13:00:00 to 2021-01-01 13:00:00
  </Component>
  <DateRangePopup
    durations={
      Array [
        Object {
          "description": "one second",
          "value": 1000,
        },
        Object {
          "description": "two seconds",
          "value": 2000,
        },
      ]
    }
    endTimeMs={1609506000000}
    format="YYYY-MM-DD HH:mm"
    isOpen={false}
    onApply={[Function]}
    onClose={[Function]}
    onNewInterval={[MockFunction]}
    startTimeMs={1609505998000}
    title="Choose Date Range?"
  />
  <div>
    <Blueprint3.Button
      className="date-range-picker__apply-button"
      data-cy="date-range-picker-edit"
      onClick={[Function]}
    >
      Edit
    </Blueprint3.Button>
  </div>
</div>
`;

exports[`DateRangePicker matches the snapshot 3`] = `
<div
  className="date-range-picker"
  id="date-range-picker-id-1"
>
  <Component
    className="bp3-label bp3-inline date-range-picker__label"
  >
    Time Range: 2021-01-01 13:00:00 to 2021-01-01 13:00:00
  </Component>
  <DateRangePopup
    durations={
      Array [
        Object {
          "description": "one second",
          "value": 1000,
        },
        Object {
          "description": "two seconds",
          "value": 2000,
        },
      ]
    }
    endTimeMs={1609506000000}
    format="YYYY-MM-DD HH:mm"
    isOpen={false}
    onApply={[Function]}
    onClose={[Function]}
    onNewInterval={[MockFunction]}
    startTimeMs={1609505998000}
    title="Choose Date Range?"
  />
  <div>
    <Blueprint3.Button
      className="date-range-picker__apply-button"
      data-cy="date-range-picker-edit"
      onClick={[Function]}
    >
      Edit
    </Blueprint3.Button>
  </div>
</div>
`;

exports[`DateRangePicker matches the snapshot 4`] = `
<div
  className="date-range-picker"
  id="date-range-picker-id-1"
>
  <Component
    className="bp3-label bp3-inline date-range-picker__label"
  >
    Time Range: 2021-01-01 13:00:00 to 2021-01-01 13:00:00
  </Component>
  <DateRangePopup
    durations={
      Array [
        Object {
          "description": "one second",
          "value": 1000,
        },
        Object {
          "description": "two seconds",
          "value": 2000,
        },
      ]
    }
    format="YYYY-MM-DD HH:mm"
    isOpen={false}
    onApply={[Function]}
    onClose={[Function]}
    onNewInterval={[MockFunction]}
    title="Choose Date Range?"
  />
  <div>
    <Blueprint3.Button
      className="date-range-picker__apply-button"
      data-cy="date-range-picker-edit"
      onClick={[Function]}
    >
      Edit
    </Blueprint3.Button>
  </div>
</div>
`;

exports[`DateRangePicker matches the snapshot 5`] = `
<div
  className="date-range-picker"
  id="date-range-picker-id-1"
>
  <Component
    className="bp3-label bp3-inline date-range-picker__label"
  >
    Time Range: 2021-01-01 13:00:00 to 2021-01-01 13:00:00
  </Component>
  <DateRangePopup
    durations={
      Array [
        Object {
          "description": "one second",
          "value": 1000,
        },
        Object {
          "description": "two seconds",
          "value": 2000,
        },
      ]
    }
    format="YYYY-MM-DD HH:mm:ss.SSS"
    isOpen={false}
    onApply={[Function]}
    onClose={[Function]}
    onNewInterval={[MockFunction]}
    title="Choose Date Range?"
  />
  <div>
    <Blueprint3.Button
      className="date-range-picker__apply-button"
      data-cy="date-range-picker-edit"
      onClick={[Function]}
    >
      Edit
    </Blueprint3.Button>
  </div>
</div>
`;

exports[`DateRangePicker matches the snapshot 6`] = `
<div
  className="date-range-picker"
  id="date-range-picker-id-1"
>
  <Component
    className="bp3-label bp3-inline date-range-picker__label"
  >
    Time Range: 2021-01-01 13:00:00 to 2021-01-01 13:00:00
  </Component>
  <DateRangePopup
    durations={
      Array [
        Object {
          "description": "one second",
          "value": 1000,
        },
        Object {
          "description": "two seconds",
          "value": 2000,
        },
      ]
    }
    format="YYYY-MM-DD HH:mm:ss"
    isOpen={false}
    onApply={[Function]}
    onClose={[Function]}
    onNewInterval={[MockFunction]}
    title="Choose Date Range?"
  />
  <div>
    <Blueprint3.Button
      className="date-range-picker__apply-button"
      data-cy="date-range-picker-edit"
      onClick={[Function]}
    >
      Edit
    </Blueprint3.Button>
  </div>
</div>
`;

exports[`DateRangePicker matches the snapshot 7`] = `
<div
  className="date-range-picker"
  id="date-range-picker-id-1"
>
  <Component
    className="bp3-label bp3-inline date-range-picker__label"
  >
    Time Range: 2021-01-01 13:00:00 to 2021-01-01 13:00:00
  </Component>
  <DateRangePopup
    durations={
      Array [
        Object {
          "description": "one second",
          "value": 1000,
        },
        Object {
          "description": "two seconds",
          "value": 2000,
        },
      ]
    }
    format="YYYY-MM-DD"
    isOpen={false}
    maxEndTimeMs={1609506000000}
    minStartTimeMs={1609505992000}
    onApply={[Function]}
    onClose={[Function]}
    onNewInterval={[MockFunction]}
    title="Choose Date Range?"
  />
  <div>
    <Blueprint3.Button
      className="date-range-picker__apply-button"
      data-cy="date-range-picker-edit"
      onClick={[Function]}
    >
      Edit
    </Blueprint3.Button>
  </div>
</div>
`;
